src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -715,23 +739,37 @@ class FeelInterpreter {
     case _ => error(EvaluationFailureType.INVALID_TYPE, s"Can't divide '$x' by '$y'")
   }
 
-  private def unaryTestExpression(x: Val)(implicit context: EvalContext): Val =
+  private def unaryTestExpression(x: Val, expression: Exp)(implicit context: EvalContext): Val =
     withVal(
       input,
       i =>
-        x match {
-          case ValBoolean(true)              => ValBoolean(true) // the expression is true
-          case ValList(ys) if ys.contains(i) =>
-            ValBoolean(true) // the expression contains the input value
-          case _ =>
-            checkEquality(i, x, _ == _, ValBoolean) match {
-              case ValBoolean(true)             => ValBoolean(true)  // the expression is the input value
-              case _ if x == ValBoolean(false)  => ValBoolean(false) // the expression is false
-              case _ if x.isInstanceOf[ValList] =>
-                ValBoolean(false) // the expression is a list but doesn't contain the input value
-              case ValNull => ValNull           // the expression can't be compared to the input value
-              case _       => ValBoolean(false) // the expression is not the input value
-            }
+        {
+          expression match {
+            case comparison: Comparison =>
+              if (comparison.x == ConstInputValue || comparison.y == ConstInputValue) {
+                x match {
+                  case ErroredVal(value, _)          => return value
+                  case _ => // Continue
+                }
+              }
+            case _ => // Continue if not a Comparison or ConstInputValue not found
+          }
```

Comment:
:x: This modification solves the case for the comparison (`? > null`). However, it doesn't solve the issue in general.

The following expressions still return `true` instead of `null`:
```
// for ? = null
? > 20 and ? < 40
// output: true 

? > 20 or ? < 40
// output: true

?
// output: true
```